synfig-studio/src/gui/modules/mod_palette/dock_paledit.cpp

```diff
@@ -537,3 +546,14 @@ Dock_PalEdit::size()const
 {
 	return palette_.size();
 }
+
+bool
+Dock_PalEdit::confirm_duplicate_color_addition()
+{
+        return App::get_ui_interface()->confirmation(
```

Comment:
I have created a separate method to be called within `add_color`. Also, does this behavior require to be controlled with User preferences or a checkbox something like: `Don't ask again`.
However, I felt it should be okay, as this won't happen regularly anyway.